src/models.ts

```diff
@@ -9,7 +9,27 @@ export enum BaseMetric {
   AVAILABILITY = 'A'
 }
 
-export const baseMetrics: ReadonlyArray<BaseMetric> = [
+export enum TemporalMetric {
+  EXPLOITABILITY = 'E',
+  REMEDIATION_LEVEL = 'RL',
+  REPORT_CONFIDENCE = 'RC'
+}
+
+export enum EnvironmentalMetric {
+  ATTACK_VECTOR = 'MAV',
+  ATTACK_COMPLEXITY = 'MAC',
+  PRIVILEGES_REQUIRED = 'MPR',
+  USER_INTERACTION = 'MUI',
+  SCOPE = 'MS',
+  CONFIDENTIALITY = 'MC',
+  INTEGRITY = 'MI',
+  AVAILABILITY = 'MA',
+  CONFIDENTIALITY_REQUIREMENT = 'CR',
+  INTEGRITY_REQUIREMENT = 'IR',
+  AVAILABILITY_REQUIREMENT = 'AR'
+}
```

Comment:
let keep naming and order as they are in spec: https://www.first.org/cvss/v3.1/specification-document#Vector-String

```suggestion
export enum EnvironmentalMetric {
  CONFIDENTIALITY_REQUIREMENT = 'CR',
  INTEGRITY_REQUIREMENT = 'IR',
  AVAILABILITY_REQUIREMENT = 'AR',
  MODIFIED_ATTACK_VECTOR = 'MAV',
  MODIFIED_ATTACK_COMPLEXITY = 'MAC',
  MODIFIED_PRIVILEGES_REQUIRED = 'MPR',
  MODIFIED_USER_INTERACTION = 'MUI',
  MODIFIED_SCOPE = 'MS',
  MODIFIED_CONFIDENTIALITY = 'MC',
  MODIFIED_INTEGRITY = 'MI',
  MODIFIED_AVAILABILITY = 'MA'
}
```